core/src/main/scala/scalaz/APair.scala

```diff
@@ -0,0 +1,38 @@
+// Copyright: 2017 - 2018 Sam Halliday
+// License: https://opensource.org/licenses/BSD-3-Clause
+
+package scalaz
+
+import scala.inline
+import scala.Some
+
+/**
+ * Type-aligned pair. Isomorphic to
+ *
+ * {{{
+ * (F[A], G[A]) forSome { type A }
+ * }}}
+ *
+ * but more robust with respect to type inference.
+ *
+ * Notation is a blend of conjunction `/\` and natural transformation `~>`.
+ */
+sealed abstract class /~\[A[_], B[_]] {
```

Comment:
I like the notation, but how is this data type related to natural transformation?

Reply:
Higher kinded, and I tend to apply natural transformations to these things a lot. i.e. I typically have a `A ~> B` to complement this.

Reply:
Then say that `~` indicates higher-kindedness. That natural transformation are often used in the vicinity of this type is a little too indirect connection 😉